refinery/core/tests.py

```diff
@@ -2041,3 +1974,66 @@ def test_collect_site_statistics_creates_new_instance(self):
         collect_site_statistics()
         self.assertEqual(initial_site_statistics_count + 1,
                          SiteStatistics.objects.count())
+
+
+class EventTests(TestCase):
+
+    def setUp(self):
+        CuserMiddleware.set_user(User.objects.create_user('testuser'))
+
+    def test_dataset_create(self):
+        create_dataset_with_necessary_models()
+        events = Event.objects.all()
+        self.assertEqual(len(events), 1)
+        self.assertRegexpMatches(
+            str(events[0]),
+            r'^\d{2}/\d{2}/\d{2} \d{2}:\d{2}:\d{2}: testuser created data set '
+            r'Test DataSet - [0-9a-f-]+$'
+        )
+
+    # DataSetPermissionsUpdateTests covers dataset_permissions_change.
+
+    def test_dataset_metadata_reupload(self):
```

Comment:
I'm always a little hesitate to have a lot of todos or empty tests...just in case responsibilities gets shifted to another person.

Reply:
I could comment these out, but that doesn't really address your concern. I think I understand why this is generally not a good idea, but here:
- We do have a pretty good roadmap of where we want to go, so I'm less hesitant to stake out a path.
- There's a lot of parallelism between different sections of the code, and if it were filled in incrementally, I'd worry that parts might be missed.
- One of my concerns about this commit is the amount of boilerplate that it will eventually require: I thought the commented out sections and the steps are a good way of expressing the work that will be needed.